tools/IceRpc.BuildTelemetry.Reporter/Program.cs

```diff
@@ -2,98 +2,92 @@
 
 using IceRpc;
 using IceRpc.BuildTelemetry;
+using Microsoft.Extensions.Logging;
 using System.Net.Security;
 using System.Reflection;
 
 var timeout = TimeSpan.FromSeconds(3); // The timeout for the complete telemetry upload process.
 
 const string uri = "icerpc://telemetry.icerpc.dev"; // The URI of the server.
 
-try
+// Create a cancellation token source to cancel the telemetry upload if it
+// takes too long.
+using var cts = new CancellationTokenSource(timeout);
+
+// Create a client connection
+var clientAuthenticationOptions = new SslClientAuthenticationOptions();
+await using var connection = new ClientConnection(new Uri(uri), clientAuthenticationOptions);
+
+// Create a reporter proxy with this client connection.
+var reporter = new ReporterProxy(connection);
+
+// Parse the IDL
+string? idl = args
+    .SkipWhile(arg => arg != "--idl")
+    .Skip(1)
+    .FirstOrDefault() ?? "unknown";
+
+// Determine the IceRPC version using the assembly version. The assembly version is kept in sync with the
+// IceRPC version.
+var assembly = Assembly.GetAssembly(typeof(SliceTelemetryData));
+string version = assembly!.GetName().Version!.ToString();
+
+// Parse the compilation hash
+string compilationHash = args
+    .SkipWhile(arg => arg != "--hash")
+    .Skip(1)
+    .FirstOrDefault() ?? "unknown";
+
+// Parse the contains-slice1 argument
+bool containsSlice1 = bool.TryParse(args
+    .SkipWhile(arg => arg != "--contains-slice1")
+    .Skip(1)
+    .FirstOrDefault(), out bool result) && result;
+
+// Parse the contains-slice2 argument
+bool containsSlice2 = bool.TryParse(args
+    .SkipWhile(arg => arg != "--contains-slice2")
+    .Skip(1)
+    .FirstOrDefault(), out result) && result;
+
+// Parse the src-file-count argument
+int sourceFileCount = int.TryParse(args
+    .SkipWhile(arg => arg != "--src-file-count")
+    .Skip(1)
+    .FirstOrDefault(), out int sourceFileCountResult) ? sourceFileCountResult : 0;
+
+// Parse the ref-file-count argument
+int referenceFileCount = int.TryParse(args
+    .SkipWhile(arg => arg != "--ref-file-count")
+    .Skip(1)
+    .FirstOrDefault(), out int referenceFileCountResult) ? referenceFileCountResult : 0;
+
+BuildTelemetry buildTelemetry = idl.ToLower() switch
```

Comment:
Given how trivial this program is, I would prefer to have separate versions for IceRpc.Protobuf.Tools and IceRpc.Slice.Tools. For a follow-up PR.

Then no need for an "idl" argument to the program.

Also note that most (maybe all) arguments are IDL-specific. --contains-slice1/--contains-slice2/--ref-file-count is Slice-only, which is not clear with the current setup.